spec/tasks/gen/action_spec.cr

```diff
@@ -2,26 +2,38 @@ require "../../spec_helper"
 
 describe Gen::Action do
   it "generates a basic action" do
-    begin
-      ARGV.push("Users::Index")
-      Gen::Action.new.call
-
-      File.exists?("src/actions/users/index.cr").should be_true
-    ensure
-      ARGV.pop
-      FileUtils.rm_rf "src/actions/users/index.cr"
+    with_cleanup do
+      io = IO::Memory.new
+      valid_action_name = "Users::Index"
+      ARGV.push(valid_action_name)
+
+      Gen::Action.new.call(io)
+      expected_path = "/src/actions/users"
+      expected_message =
+        "Done generating #{valid_action_name.colorize(:green)}" \
+        " in #{expected_path.colorize(:green)}"
```

Comment:
Maybe instead of testing the exact message you can use `contain` to test that is has the right path and action:

```crystal
io.to_s.should contain(valid_action_name)
io.to_s.should contain("/src/actions/users")
```

That way we can change the message without breaking the test

Reply:
I'm a huge fan of using contain throughout the specs for these messages. 👍